### DISCUSSION TITLE
Pre-save file hook

### DISCUSSION THREAD

**Discussion by benlau** [microsoft/vscode-jupyter#15445] [Questions and Answers]
Jupyter Notebook/Lab reads $JUPYTER_CONFIG_DIR/jupyter_notebook_config.py on startup. Users may add a pre-save hook to modify the content to be saved.

For example, I usually use it to remove the execution count to make it more VCS-friendly.

However, in VSCode, Jupyter doesn't seem to read from jupyter_notebook_config.py. Is there any alternative method that could register a pre-save file hook in VSCode Jupyter?

Thanks.

**Comment by iso2013** [comment]
I have the same issue. Notably, it also doesn't seem to respect the config if you use it through an existing server - my guess is it's not the underlying server that saves the file but some other VSCode package.
